observability/tracing.go

```diff
@@ -0,0 +1,54 @@
+// The observability package provides utilities for the Kiali server
+// to instrument itself with observability tools such as tracing to provide
+// better insights into server performance.
+package observability
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+)
+
+const (
+	// TracingService is the name of the kiali tracer service.
+	TracingService = "kiali-server"
```

Comment:
This shouldn't be a constant - I think - it should have some contextual information about cluster/mesh, to identify the Kiali server, perhaps "kiali-server.istio-system" ? I was thinking in federated / multi-mesh / multi-cluster ? scenarios.

Perhaps a kiali-server.east-mesh-system.clusterA, would be nice, and then it would be prepared for the future.

Reply:
That's a good suggestion and is more in line with how other services are named by istio by default. It looks like the namespace Kiali is deployed inside of is included in the config so that would be easy to add. The mesh/cluster id might be harder to determine. Kiali may not even be a part of any mesh since it's normal to run Kiali without a sidecar. The cluster id might be able to be fetched from istio or some kube api.